junebug/channel.py

```diff
@@ -29,6 +30,13 @@ class InvalidChannelType(JunebugError):
     'xmpp': 'vumi.transports.xmpp.XMPPTransport',
 }
 
+allowed_message_fields = [
+    'transport_name', 'timestamp', 'in_reply_to', 'to_addr', 'from_addr',
+    'content', 'session_event', 'helper_metadata', 'message_id']
+# excluded fields: from_addr_type, group, provider, routing_metadata,
+# to_addr_type, from_addr_type, message_version, transport_metadata,
+# message_type, transport_type
```

Comment:
If junebug should also support sending reply messages, I think we might need transport metadata. I think some transports add some metadata for inbound messages that reply messages use, like session identifiers.

Is `message_id` maybe something we want to generate ourselves.

Is `transport_name` something we need from people?

Any reason to disallow `group`?